client.go

```diff
@@ -864,12 +864,12 @@ func (c *Client) MkdirAll(path string) error {
 
 	// Slow path: make sure parent exists and then call Mkdir for path.
 	i := len(path)
-	for i > 0 && os.IsPathSeparator(path[i-1]) { // Skip trailing path separator.
+	for i > 0 && path[i-1] == '/' { // Skip trailing path separator.
 		i--
 	}
```

Comment:
Hm… we could probably just make use of `path.Split(…)` instead, rather than doing this raw byte search ourselves. This is probably fine as is though.